src/SearchResults.js

```diff
@@ -0,0 +1,79 @@
+// Create a `<SearchResults />` component that shows hotel bookings in a `<table>`
+// element. Each booking will have an id, title, first name, surname, email, room id,
+// check in date and check out date. You can make up data to show in the table.
+// Then show `<SearchResults />` component within the `<Bookings />` component that is provided for you. Be sure to split out your components into small well-named components, similar to the method used in exercise 1. Hint: You will find some useful `<table>` examples in the [Bootstrap documentation for tables](https://getbootstrap.com/docs/4.2/content/tables/#examples)
+
+import React, { Component } from 'react'
+var moment = require('moment')
+const selectedRowIdDefault = 0
+class SearchResults extends Component {
+  state = {
+    rowSelected: false,
+    selectedRowId: selectedRowIdDefault
+  }
+  handleSelection = selectedRowId => {
+    this.setState(previousState => {
+      return {
+        selectedRowId: selectedRowId
+      }
+    })
+  }
+
+  toggle = e => {
+    const indexRaw = e.target.parentNode.id
+    const index = parseInt(indexRaw, 10)
+
+    this.handleSelection(Number.isNaN(index) ? selectedRowIdDefault : index)
+  }
+
+  render () {
+    return (
+      <table className='table'>
+        <thead className='thead-dark'>
+          <tr>
+            <th scope='col'>#</th>
+            <th scope='col'>Title</th>
+            <th scope='col'>First Name</th>
+            <th scope='col'>Surname</th>
+            <th scope='col'>Email</th>
+            <th scope='col'>Room Id</th>
+            <th scope='col'>Check-in Date</th>
+            <th scope='col'>Check-out Date</th>
+            <th scope='col'>Duration of Stay</th>
+          </tr>
+        </thead>
+        <tbody>
+          {this.props.results.map(row => {
+            return (
+              <tr
+                id={row.id}
+                key={row.id}
+                onClick={this.toggle}
+                className={
+                  this.state.selectedRowId === row.id ? 'bg-info' : 'bg-light'
```

Comment:
Really great job! Well done figuring this out - it is quite a tricky problem :)

However there is actually a much simpler approach to this problem. Do you remember in class we had the example with multiple Counter components? Each counter remembered their own count.

We can do a similar thing here - if we extracted a table row component (one for each result), then they could each remember their own state. In fact they would only need to remember if they were selected or not (which sounds quite like a boolean to me 😉). How would we go about implementing that?